<script>
  // no use async/await
  // const getData1 =  () => {
  //   const data =   fetch('https://jsonplaceholder.typicode.com/photos');
  //   console.log('------data 1-------')
  //   console.log(data);
  //   console.log('---end of data 1---');
  // }

  // //  use async/await
  // const getData2 = async () => {
  //   const data =  await fetch('https://jsonplaceholder.typicode.com/photos');
  //   console.log('------data 2-------')
  //   console.log(data);
  //   console.log('---end of data 2---');
  // }
  // getData1();
  // getData2();

  // const promise = new Promise(async (resolve, reject) => {
  //   const data = await fetch('https://jsonplaceholder.typicode.com/photos')
  //     //.then(data => console.log(data));
  //   console.log(data.status)
  //   if(data.status === 200) {
  //     resolve(data);
  //   }
  //   else {
  //     reject('error');
  //   }
  // })
  //   .then((data) => {
  //       console.log('Success!')
  //       console.log(data);

  //   })
  //   .catch((error) => {
  //       console.log(error)
  //   })

  const promise = new Promise(async (resolve, reject) => {
    const data = await fetch("https://jsonplaceholder.typicode.com/photos")
      .then((data) => {
        if (data.status === 200) {
          resolve(data);
        } 
        else {
          reject("error");
        }
      });
  })
    .then((data) => {
      console.log("Success!");
      console.log(data);
    })
    .catch((error) => {
      console.log(error);
    });

</script>
